nsxt/gateway_common.go

```diff
@@ -271,6 +271,50 @@ func initChildLocaleService(serviceStruct *model.LocaleServices, markForDelete b
 	return dataValue.(*data.StructValue), nil
 }
 
+func policyGatewayLocalServiceChanged(d *schema.ResourceData) bool {
```

Comment:
There seem to be a fix upstream for this https://github.com/hashicorp/terraform-plugin-sdk/pull/362. Not sure how to test this as it seems to be maintenance branch. But perhaps worth checking this out as this code is complex..

Reply:
According to this https://github.com/hashicorp/terraform-plugin-sdk/commit/83ea1db2e126b6b3bf2bd1f6a367367034359b2f#diff-4ac32a78649ca5bdd8e0ba38b7006a1eL35 this fix should be in plugin sdk 0.12.

Reply:
I've tested HasChange with maps in the provider and it works!

Reply:
It didn't work for me on my setup, but maybe something is not up to date here. I'll update the code. Thanks.